beamer/agent/chain.py

```diff
@@ -611,73 +615,76 @@ def maybe_prove(claim: Claim, context: Context) -> None:
     if not can_prove:
         return
 
-    if request.fill_tx is not None:
-        proof_tx = request.fill_tx
-    elif claim.invalidation_tx is not None:
-        proof_tx = claim.invalidation_tx
-
-    if proof_tx in context.l1_resolutions:
+    if proved_tx in context.l1_resolutions:
         return
 
+    assert proved_tx is not None
+
     future = context.task_pool.submit(
         run_relayer_for_tx,
         context.config.base_chain_rpc_url,
         context.target_chain.rpc_url,
         context.source_chain.rpc_url,
         context.config.account.key,
-        proof_tx,
+        proved_tx,
         True,
     )
 
     def on_future_done(f: Future) -> None:
         try:
             f.result()
         except Exception as ex:
-            context.logger.error("Optimism prove failed", ex=ex, tx_hash=proof_tx)
+            context.logger.error("Optimism prove failed", ex=ex, tx_hash=proved_tx)
         else:
-            claim.message_proved = True
+            assert request is not None
+            claim.proved_tx = proved_tx
+            if request.fill_tx == proved_tx:
+                request.fill_timestamp = Timestamp(int(time.time()))
+            elif claim.invalidation_tx == proved_tx:
+                claim.invalidation_timestamp = Timestamp(int(time.time()))
         finally:
-            context.l1_resolutions.pop(proof_tx, None)
+            assert proved_tx is not None
```

Comment:
Do we need this assert? `proved_tx` is guaranteed not to be `None` here.

Reply:
same answer as above

Reply:
Argument 1 to "pop" of "dict" has incompatible type "Optional[HexBytes]"; expected "HexBytes"